lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -252,17 +259,22 @@ private static LockServiceImpl create(LockServerOptions options, Ownable<Executo
         return lockService;
     }
 
-    private LockServiceImpl(LockServerOptions options, Runnable callOnClose, Ownable<ExecutorService> executor) {
+    /**
+     * Only {@link LockServerOptions#threadInfoConfiguration()} is actually reloadable
+     */
+    private LockServiceImpl(
+            Refreshable<LockServerOptions> options, Runnable callOnClose, Ownable<ExecutorService> executor) {
+        LockServerOptions currentOptions = options.current();
         this.lockReapRunner = new LockReapRunner(executor);
         this.callOnClose = callOnClose;
-        this.isStandaloneServer = options.isStandaloneServer();
-        this.maxAllowedLockTimeout = SimpleTimeDuration.of(options.getMaxAllowedLockTimeout());
-        this.maxAllowedClockDrift = SimpleTimeDuration.of(options.getMaxAllowedClockDrift());
-        this.maxNormalLockAge = SimpleTimeDuration.of(options.getMaxNormalLockAge());
-        this.stuckTransactionTimeout = SimpleTimeDuration.of(options.getStuckTransactionTimeout());
-        this.slowLogTriggerMillis = options.slowLogTriggerMillis();
-        this.threadInfoSnapshotManager =
-                new LockThreadInfoSnapshotManager(options.threadInfoConfiguration(), () -> heldLocksTokenMap);
+        this.isStandaloneServer = currentOptions.isStandaloneServer();
+        this.maxAllowedLockTimeout = SimpleTimeDuration.of(currentOptions.getMaxAllowedLockTimeout());
+        this.maxAllowedClockDrift = SimpleTimeDuration.of(currentOptions.getMaxAllowedClockDrift());
+        this.maxNormalLockAge = SimpleTimeDuration.of(currentOptions.getMaxNormalLockAge());
+        this.stuckTransactionTimeout = SimpleTimeDuration.of(currentOptions.getStuckTransactionTimeout());
+        this.slowLogTriggerMillis = currentOptions.slowLogTriggerMillis();
+        this.threadInfoSnapshotManager = new LockThreadInfoSnapshotManager(
+                options.map(LockServerOptions::threadInfoConfiguration), () -> heldLocksTokenMap);
```

Comment:
Let's remove/collapse start into the constructor; typically I'd recommend doing this inside a static constructor, but seeing as we do this for the lock reaper presently, its ok